src/array/cpu/array_op_impl.cc

```diff
@@ -51,10 +53,9 @@ IdArray BinaryElewise(IdArray lhs, IdArray rhs) {
   IdType* ret_data = static_cast<IdType*>(ret->data);
   // TODO(BarclayII): this usually incurs lots of overhead in thread spawning, scheduling,
   // etc., especially since the workload is very light.  Need to replace with parallel_for.
-// #pragma omp parallel for
-  for (int64_t i = 0; i < lhs->shape[0]; ++i) {
+  parallel_for(0, lhs->shape[0], [=](size_t i){
```

Comment:
From my experience this kind of elementwise operations needs a bigger grain size.  A grain size of 1 incurs a lot of overhead on small graph operations.

Reply:
I haven't noticed any slowdown in our internal CI after these changes. But this piece of code could be a good candidate for further performance improvements using `parallel_for` with grain size.

Is it ok if I open an issue with links to codes that are potential candidates for improvements with grain size?

Reply:
There is an issue talking about our experience with OpenMP in https://github.com/dmlc/dgl/issues/2408.

We specifically disabled the OpenMP loop because we indeed observed slowdown especially on small graphs.  Our benchmark is in https://github.com/dmlc/dgl/pull/2412.

Reply:
Judging from benchmark results, you're right that grain size should be set. I will look at it. Do you have any initial suggestions for the grain size value?

Reply:
I could not find how PyTorch handles such a light workload as they seem to use AVX.  I did find an example of a big grain size though:
```C
    at::parallel_for(0, nnz, 10000, [&](int64_t start, int64_t end) {
      int64_t h, hp0, hp1;
      for (auto i = start; i < end; i++) {
        hp0 = indices[i];
        hp1 = (i+1 == nnz) ?  dim : indices[i+1];
        if (hp0 != hp1) {
          for (h = hp0; h < hp1; h++) {
            csr_accessor[h+1] = i+1;
          }
        }
      }
    });
```
Maybe we can get a sense of the relationship between grain size and workload there.  For simple binary arithmetic I would imagine the grain size to be significantly larger (maybe 1000000?) since it can benefit from vectorized instructions and caches already.